wolfssl/test.h

```diff
@@ -3925,10 +3925,70 @@ static WC_INLINE const char* mymktemp(char *tempfn, int len, int num)
 }
 
 
+#ifdef HAVE_SESSION_TICKET
 
-#if defined(HAVE_SESSION_TICKET) && \
-    ((defined(HAVE_CHACHA) && defined(HAVE_POLY1305)) || \
-      defined(HAVE_AESGCM))
+    static THREAD_LS_T WC_RNG myKey_rng;
+
+    static WC_INLINE void TicketCleanup(void)
+    {
+        wc_FreeRng(&myKey_rng);
+    }
+
+#ifdef OPENSSL_EXTRA
+
+    typedef struct key_ctx {
+        byte name[WOLFSSL_TICKET_NAME_SZ]; /* server name */
+        byte key[AES_256_KEY_SIZE]; /* cipher key */
+        byte hmacKey[WOLFSSL_TICKET_NAME_SZ]; /* hmac key */
+        byte iv[WOLFSSL_TICKET_IV_SZ]; /* cipher iv */
+    } key_ctx;
+
+    static THREAD_LS_T key_ctx myKey_ctx;
+
+    static WC_INLINE int TicketInit(void)
+    {
+        int ret = wc_InitRng(&myKey_rng);
+        if (ret != 0) return ret;
+
+        ret = wc_RNG_GenerateBlock(&myKey_rng, myKey_ctx.name, sizeof(myKey_ctx.name));
+        if (ret != 0) return ret;
+
+        ret = wc_RNG_GenerateBlock(&myKey_rng, myKey_ctx.key, sizeof(myKey_ctx.key));
+        if (ret != 0) return ret;
+
+        ret = wc_RNG_GenerateBlock(&myKey_rng, myKey_ctx.hmacKey, sizeof(myKey_ctx.hmacKey));
+        if (ret != 0) return ret;
+
+        ret = wc_RNG_GenerateBlock(&myKey_rng, myKey_ctx.iv,sizeof(myKey_ctx.iv));
+        if (ret != 0) return ret;
+
+        return 0;
+    }
+
+    static WC_INLINE int myTicketEncCbOpenSSL(WOLFSSL* ssl,
```

Comment:
Don't want this.
OpenSSL's encryption is not needed for comparability - don't need to share the keys between us.
The wolfSSL callback is want we to be using. In particular the one I'm adding.

Reply:
This is done to test the `wolfSSL_CTX_set_tlsext_ticket_key_cb` API especially the `wolfSSL_TicketKeyCb` wrapper function. Without this there is no unit testing on this code.

Reply:
Does Nginx set a callback?

Reply:
It does when the user specifies the `ssl_session_ticket_key` directive. Otherwise it relies on a default ticket implementation.

Reply:
Tell me what ssl_session_ticket_key does in Nginx to need that!
Does Nginx try to parse out the data?

Reply:
The `ssl_session_ticket_key` directive allows users to set the key that is used to encrypt and decrypt tickets. This function is testing an OpenSSL style callback.
Nginx does not parse out the data.

Reply:
How does it set the key?
Can we replace the implementation with out code when compiled to work with wolfSSL?

Reply:
The key is set by the ticket callback called in `wolfSSL_TicketKeyCb`. The function is `ngx_ssl_session_ticket_key_callback` in Nginx.

Reply:
So Nginx is implementing OpenSSL's callback.
You want test that setting the OpenSSL callback works with our code - put it into api.c.
Don't want the client/server programs using this API.

Reply:
Moved.

Reply:
Thank you.